src/features/postsSlice.tsx

```diff
@@ -0,0 +1,63 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getUserPosts } from '../api/posts';
+import { Post } from '../types/Post';
+
+export interface T {
+  loaded: boolean,
+  hasError: boolean,
+  items: Post[],
+}
+
+export const initialState: T = {
+  loaded: false,
+  hasError: false,
+  items: [] as Post[],
+};
+
+export const init = createAsyncThunk(
+  'posts/loadUserPosts',
+  async (userId: number) => {
+    const post = await getUserPosts(userId);
+
+    return post;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+  },
+  extraReducers: (builder) => {
+    // builder.addCase(init.pending, (state) => {
+    //   return { ...state, loaded: true };
+    // });
+    // builder.addCase(init.fulfilled, (state, action) => {
+    //   return { ...state, loaded: false, items: action.payload };
+    // });
+    // builder.addCase(init.rejected, (state) => {
+    //   return { ...state, loaded: false, hasError: true };
+    // });
+
+    // builder.addCase(init.rejected, (state) => ({
+    //   state.loaded = false;
+    //   state.hasError = true;
+    // }));
+
+    /* eslint-disable no-param-reassign */
```

Comment:
bad idea to disable eslint rules

Reply:
If I remove this I get this error. The mentor said to use this

`Assignment to property of function parameter 'state'.eslint[no-param-reassign](https://eslint.org/docs/rules/no-param-reassign)
(parameter) state: WritableDraft<T>`